### ISSUE TITLE
Cannot copy out of meta tensor; no data! with sd_turbo 1.5

### ISSUE THREAD

**Issue by olegchomp** [huggingface/diffusers#8578]
### Describe the bug

Error on loading from single file. Might be related to this one https://huggingface.co/stabilityai/sd-turbo/discussions/1

### Reproduction

```
StableDiffusionPipeline.from_single_file(
                    'sd_turbo.safetensors',
                    cache_dir = 'models/checkpoints', 
                    use_safetensors=True,
                    local_files_only=False,
                    torch_dtype = torch.float16,
                    variant="fp16",
                    safety_checker=None
                ).to(device=self.device, dtype=self.dtype)
```

### Logs

```
warnings.warn(
Some weights of the model checkpoint were not used when initializing CLIPTextModel:
 ['text_model.embeddings.position_ids']
 
Cannot copy out of meta tensor; no data!
```

### System Info

- `diffusers` version: 0.27.2 (tried also with latest)
- Platform: Windows-10-10.0.22631-SP0
- Python version: 3.11.1
- PyTorch version (GPU?): 2.1.0+cu118 (True)
- Huggingface_hub version: 0.23.4
- Transformers version: 4.41.2
- Accelerate version: 0.31.0
- xFormers version: not installed

### Who can help?

 @yiyixuxu @DN6 @sayakpaul

**Comment by sayakpaul** [comment]
Could you provide a link to the safetensors file too? Cc: @DN6

**Comment by olegchomp** [comment]
> Could you provide a link to the safetensors file too? Cc: @DN6 

https://huggingface.co/stabilityai/sd-turbo/blob/main/sd_turbo.safetensors

**Comment by DN6** [comment]
Should be fixed with : #8462
